### PR TITLE
Parameters::isSet: pass parameter struct instead of stringified

### PR DESCRIPTION
**Pull request by akva2**
Let's get rid of the final method that is string based.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-simulators=5282 please

File: opm/models/utils/parametersystem.hh [hunk 1]
```diff
@@ -880,15 +880,18 @@ public:
...
      * If the parameter in question has not been registered, this throws an exception.
      */
-    template <class ParamType>
-    static bool isSet(const char* paramName,
-                      bool errorIfNotRegistered = true)
+    template <template<class,class> class Param>
+    static bool isSet(bool errorIfNotRegistered = true)
     {
-
+        const std::string paramName = getPropName<TypeTag,Param>();
```
Comment (bska) [inline opm/models/utils/parametersystem.hh:886]: If you're explicitly storing this in a `std::string`, then I suppose we don't need [`canonicalName`](https://github.com/OPM/opm-models/blob/769bb55c688eb09481295d32dbce6b79b9eaaa25/opm/models/utils/parametersystem.hh#L905) anymore and might as well pass `paramName` directly as an argument to [`hasKey()`](https://github.com/OPM/opm-models/blob/769bb55c688eb09481295d32dbce6b79b9eaaa25/opm/models/utils/parametersystem.hh#L908).

**Review by bska — APPROVED** [review]
This looks good to me and I'll be happy to merge it in its current form, but I think there's one more simplification we can make here.

File: opm/models/utils/parametersystem.hh (hunk 1 above)
Comment (akva2) [reply opm/models/utils/parametersystem.hh:886]: indeed.

**Comment by akva2** [comment]
jenkins build this opm-simulators=5282 please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me and I'll merge into master.

**Pull request merged**